src/SprykerEco/Zed/Unzer/Business/ApiAdapter/Mapper/UnzerAuthorizePaymentMapper.php

```diff
@@ -109,4 +111,25 @@ public function mapUnzerApiAuthorizeResponseTransferToUnzerPaymentTransfer(
             ->setPaymentResource((new UnzerPaymentResourceTransfer())->setId($unzerApiAuthorizeResponseTransfer->getTypeId()))
             ->setRedirectUrl($unzerApiAuthorizeResponseTransfer->getRedirectUrl());
     }
+
+    /**
+     * @param \Generated\Shared\Transfer\UnzerApiErrorResponseTransfer $unzerApiErrorResponseTransfer
+     * @param \Generated\Shared\Transfer\UnzerPaymentTransfer $unzerPaymentTransfer
+     *
+     * @return \Generated\Shared\Transfer\UnzerPaymentTransfer
+     */
+    public function mapUnzerApiErrorResponseTransferToUnzerPaymentTransfer(
+        UnzerApiErrorResponseTransfer $unzerApiErrorResponseTransfer,
+        UnzerPaymentTransfer $unzerPaymentTransfer
+    ): UnzerPaymentTransfer {
+        foreach ($unzerApiErrorResponseTransfer->getErrors() as $unzerApiResponseErrorTransfer) {
+            $unzerPaymentTransfer->addError(
+                (new UnzerPaymentErrorTransfer())
+                    ->setMessage($unzerApiResponseErrorTransfer->getCustomerMessage())
+                    ->setErrorCode((int)$unzerApiResponseErrorTransfer->getCode()),
```

Comment:
Would be nice to extract it to a separate method.

Reply:
Fixed.